### PR TITLE
replaced spaces with commas for setting passenv

### PR DESCRIPTION
**Pull request by a-dubs**
## Proposed Commit Message
replaced spaces with commas for setting passenv in tox.ini

```
Replaced spaces with commas for setting passenv in tox.ini  

A recent update of tox-ini made a breaking change in the `tox.ini` file
when using whitespace to set multiple values to the `passenv` variable.
This change was believed to have occurred on the following commit
in the tox repository:
https://github.com/tox-dev/tox/commit/f4840a03
```

## Checklist:

 - [ ] My code follows the process laid out in [the documentation](https://cloudinit.readthedocs.io/en/latest/development/contributing.html)
 - [ ] I have updated or added any unit tests accordingly
 - [ ] I have updated or added any documentation accordingly

### CODE REVIEW

**Review by blackboxsw — CHANGES_REQUESTED** [review]
Thanks @a-dubs, per https://cloudinit.readthedocs.io/en/latest/development/contributing.html please also add your github username to `tools/.github-cla-signers` to allow CI to be happy.

**Comment by blackboxsw** [comment]
@a-dubs Also please add a link to the upstream tox commit we think changed this behavior somewhere in your suggested commit message at the top of this PR description 
https://github.com/tox-dev/tox/commit/f4840a03

**Review by blackboxsw — APPROVED** [review]
LGTM, weclome to cloud-init @a-dubs . Just waiting on CI

File: tox.ini [hunk 1]
```diff
@@ -280,7 +280,7 @@ setenv =
 allowlist_externals = sh
 commands = sh -c "{envpython} -m pytest --log-cli-level=INFO -vv {posargs:tests/integration_tests/none} || [ $? -eq 1 ]"
 deps = {[testenv:integration-tests]deps}
-passenv = *_proxy CLOUD_INIT_* PYCLOUDLIB_* SSH_AUTH_SOCK OS_* GOOGLE_* GCP_*
+passenv = *_proxy CLOUD_INIT_*,PYCLOUDLIB_*,SSH_AUTH_SOCK,OS_*,GOOGLE_*,GCP_*
```
Comment (aciba90) [inline tox.ini:283]:
```suggestion
passenv = *_proxy,CLOUD_INIT_*,PYCLOUDLIB_*,SSH_AUTH_SOCK,OS_*,GOOGLE_*,GCP_*
```

**Review by aciba90 — CHANGES_REQUESTED** [review]

File: tox.ini (hunk 1 above)
Comment (a-dubs) [reply tox.ini]: fixed with commit [7c733cd](https://github.com/canonical/cloud-init/pull/4269/commits/7c733cd716556e29cb3ba28305eff4d674fd8c56)

**Review by aciba90 — CHANGES_REQUESTED** [review]
We can see in integration job that the env vars are not being correctly passed:

[This line](https://github.com/canonical/cloud-init/actions/runs/5650634473/job/15307481227?pr=4269#step:7:7) defines a value for CLOUD_INIT_CLOUD_INIT_SOURCE which not passed to the inner tox environment, see https://github.com/canonical/cloud-init/actions/runs/5650634473/job/15307481227?pr=4269#step:7:25

The problem is that the integration test are run in Jammy which contains tox 3, and the comma separated notation for `passenv` is tox 4 only.

One way to solve this to use separate the env vars by new line, which is compatible with tox 3 and 4. More info: https://tox.wiki/en/4.6.4/upgrading.html#changed-ini-rules

Here is a patch containing the porpoused change:

```diff
diff --git a/tox.ini b/tox.ini
index db9082b3d..083c1e4c6 100644
--- a/tox.ini
+++ b/tox.ini
@@ -265,12 +265,20 @@ commands = {envpython} -m pytest -vv \
        {posargs:tests/integration_tests}
 deps =
     -r{toxinidir}/integration-requirements.txt
-passenv = CLOUD_INIT_*,PYCLOUDLIB_*,SSH_AUTH_SOCK,OS_*
+passenv =
+    CLOUD_INIT_*
+    PYCLOUDLIB_*
+    SSH_AUTH_SOCK
+    OS_*
 
 [testenv:integration-tests-ci]
 commands = {[testenv:integration-tests]commands}
 deps = {[testenv:integration-tests]deps}
-passenv = CLOUD_INIT_*,SSH_AUTH_SOCK,OS_*,TRAVIS
+passenv =
+    CLOUD_INIT_*
+    SSH_AUTH_SOCK
+    OS_*
+    TRAVIS
 setenv =
     PYTEST_ADDOPTS="-m ci and not adhoc"
 
@@ -280,7 +288,14 @@ setenv =
 allowlist_externals = sh
 commands = sh -c "{envpython} -m pytest --log-cli-level=INFO -vv {posargs:tests/integration_tests/none} || [ $? -eq 1 ]"
 deps = {[testenv:integration-tests]deps}
-passenv = *_proxy,CLOUD_INIT_*,PYCLOUDLIB_*,SSH_AUTH_SOCK,OS_*,GOOGLE_*,GCP_*
+passenv =
+    *_proxy
+    CLOUD_INIT_*
+    PYCLOUDLIB_*
+    SSH_AUTH_SOCK
+    OS_*
+    GOOGLE_*
+    GCP_*
 setenv =
     PYTEST_ADDOPTS="-m not adhoc"
```

**Review by aciba90 — APPROVED** [review]
+1, thanks and welcome to cloud-init!

**Pull request merged**